core/src/main/java/hudson/tasks/LogRotator.java

```diff
@@ -108,8 +109,44 @@ public LogRotator(int daysToKeep, int numToKeep, int artifactDaysToKeep, int art
         
     }
 
+    /**
+     * This method calls {@link Run#delete()} with all checked exceptions
+     * properly caught and reported in the log.
+     * 
+     * @param r the run to delete, must not be null.
```

Comment:
```suggestion
     * @param r the run to delete, must not be null.
     * @since TODO
```

Protected API of a public class is a public API

Reply:
Done